lms/views/api/canvas/pages.py

```diff
@@ -43,11 +51,72 @@ def list_pages(self):
 
     @view_config(request_method="GET", route_name="canvas_api.pages.via_url")
     def via_url(self):
-        application_instance = self.request.lti_user.application_instance
+        course_copy_plugin = self.request.product.plugin.course_copy
+        current_course = self.request.find_service(name="course").get_by_context_id(
+            self.request.lti_user.lti.course_id, raise_on_missing=True
+        )
+        current_course_id = str(
+            current_course.extra["canvas"]["custom_canvas_course_id"]
+        )
         assignment = self.request.find_service(name="assignment").get_assignment(
-            application_instance.tool_consumer_instance_guid,
+            self.request.lti_user.application_instance.tool_consumer_instance_guid,
             self.request.lti_user.lti.assignment_id,
         )
+        document_url = assignment.document_url
+        document_course_id, document_page_id = self._parse_document_url(document_url)
+
+        effective_page_id = None
+        if current_course_id == document_course_id:
```

Comment:
If the course in the document_url is the same as in the launch, just use the info from the document_url. No course copy here.